### DISCUSSION TITLE
AttributeError: 'CRNN' object has no attribute 'load_state_dict'

### DISCUSSION THREAD

**Discussion by Nithinkumar-29** [mindee/doctr#1456] [Q&A] [answered]
I have trained a crnn_vgg16_bn on a custom dataset and tried to load the model checkpoints I saved as .pt files and it is giving the following error

AttributeError: 'CRNN' object has no attribute 'load_state_dict'

The following is the piece of code I wrote to load the .pt files:
def initialize_handwritten_models():
  det_model = db_resnet50(pretrained=True)
  reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False,vocab=Vocabs["french])
  reco_dict = torch.load("/path/to/pt file",map_location="cpu")
  reco_model.load_state_dict(reco_dict)
  det_predictor = DetectionPredictor(
    PreProcessor(
        (1024, 1024),
        batch_size=1,
        mean=(0.798, 0.785, 0.772),
        std=(0.264, 0.2749, 0.287)
    ),
    det_model
  )

  reco_predictor = RecognitionPredictor(
    PreProcessor(
        (32, 128),
        preserve_aspect_ratio=True,
        batch_size=32,
        mean=(0.694, 0.695, 0.693),
        std=(0.299, 0.296, 0.301)
    ),
    reco_model
  )

  predictor = OCRPredictor(det_predictor, reco_predictor)
  
  return predictor

**Comment by felixdittrich92** [comment, marked as answer]
Hi @Nithinkumar-29 you have installed both backends Tf and PyTorch ? 
If so put `USE_TORCH=1`  in front of your command -> `USE_TORCH=1 python3 ...`

**Reply by Nithinkumar-29** [reply]
Hi, Thank you for your response. I'm operationg on colab so I'm running cells and not running any commands. So can you please guide me keeping in view that my code is in cells rather than python files

Thank you for your time

**Reply by felixdittrich92** [reply]
You can also write in the first cell before doctr imports: 
```
os.environ['USE_TORCH'] = '1'
```

For example:

```
import os
os.environ['USE_TORCH'] = '1'

from doctr.io import DocumentFile
from doctr.models import ocr_predictor
```

**Reply by Nithinkumar-29** [reply]
Hi, I have tried the above suggestion but it didn't fix the issue.
Here are my import statements if it is of any help:
import warnings
warnings.filterwarnings("ignore")

import torch
import os
os.environ['USE_TORCH'] = '1'
from PIL import Image
from doctr.io import DocumentFile
from doctr.models import ocr_predictor
from doctr.models import crnn_vgg16_bn, db_resnet50
from doctr.models.predictor import OCRPredictor
from doctr.models.detection.predictor import DetectionPredictor
from doctr.models.recognition.predictor import RecognitionPredictor
from doctr.models.preprocessor import PreProcessor

**Reply by Nithinkumar-29** [reply]
Your answer was right installing only the torch version using the command: pip install "python-doctr[torch]" solved the error.
Thank you for your time and assistance.
